hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -295,29 +294,39 @@ object HoodieBaseRelation {
                            filters: Seq[Filter],
                            options: Map[String, String],
                            hadoopConf: Configuration): PartitionedFile => Iterator[InternalRow] = {
-    val hfileReader = createHFileReader(
```

Comment:
The reason why it was called here as opposed to call w/in the returned lambda -- is to avoid serialization of all these passed in parameters down to Spark executor. This is likely the reason why your build is failing at the moment.

Let's keep it as is, and please add a comment elaborating why this should be done in here (and executed w/in Spark Driver)

Reply:
Thanks for your advise 👍 

Modified according to your comments, PTAL